public_dropin_environments/python3_sklearn_fedramp_compliant/Dockerfile

```diff
@@ -0,0 +1,56 @@
+# This is a private chain-guard development image that is stored in DataRobot's private registry.
+# Replace it with your own development chain-gaurd image if you build your own.
+ARG BASE_ROOT_IMAGE=datarobotdev/mirror_chainguard_datarobot.com_python-fips:3.11-dev
+FROM ${BASE_ROOT_IMAGE} AS build
+
+# This is a private production chain-guard image that is stored in DataRobot's private registry.
+# Replace it with your own production chain-gaurd image if you build your own.
+FROM datarobotdev/mirror_chainguard_datarobot.com_python-fips:3.11
+
+USER root
+
+# Required to run the entrypoint script
+COPY --from=build /bin/sh /bin/sh
+
+# Required to change the ownership of copied files into the managed-image
+COPY --from=build /bin/chown /bin/chown
+
+# Required to change the permissions of the 'start_server.sh' that is copied into the managed-image
+COPY --from=build /bin/chmod /bin/chmod
+
+# Required for DR backend to create /opt/code (Maybe required by applications other than custom-models)
+COPY --from=build /bin/mkdir /bin/mkdir
+
+# Required for custom-models to install dependencies
+COPY --from=build /usr/bin/pip /usr/bin/pip
+
+# Just for convenience
+COPY --from=build /bin/ls /bin/ls
```

Comment:
FWIW, I posted a question to ChainGuard about this approach of copying these binaries from dev image to production image to make sure this isn't an anti-pattern or anything (https://datarobot.slack.com/archives/C07LR41TMV2/p1738692838032889). I'm only curious because I think it is a clever approach, yet I didn't see them recommending it in any of their documentation I have read. Instead I simply see them talk about using the `-dev` image in production if you need to customize the image 🤷‍♂️

Reply:
tl;dr: this is one of the two approved approaches and IMO, for this usecase, the better of the two but I can tag you in the thread so you are familiar with the other approach as well.